File: src/test/testData/java/implicit/LinkedList.java
package ru.ustinov;

import java.net.URL;
import java.util.HashSet;
import java.util.LinkedList;

class Simple {
    public static void main() {
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">LinkedList<URL> a;</warning>
        LinkedList<Integer> b;
        <warning descr="This data structure can probably implicitly invoke URL#hashCode()">LinkedList<HashSet<URL>> c, d;</warning>
        LinkedList<HashSet<Integer>> e, f;
    }
}

File: src/test/testData/kotlin/implicit/LinkedList.kt
import java.net.URL
import java.util.HashSet
import java.util.LinkedList

@Suppress("UNUSED_VARIABLE")
fun main() {
    <warning descr="This data structure can probably implicitly invoke URL#hashCode()">val a: LinkedList<URL></warning>
    val b: LinkedList<Int>
    <warning descr="This data structure can probably implicitly invoke URL#hashCode()">val c: LinkedList<HashSet<URL>></warning>
    val d: LinkedList<HashSet<Int>>
}